lib/u2/ScrollDAOTable.es6.js

```diff
@@ -0,0 +1,208 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  name: 'ScrollDAOTable',
+  package: 'org.chromium.apis.web',
+  extends: 'foam.u2.View',
+
+  documentation: `Decorator for foam.u2.view.ScrollDAOView that adds column
+      headers, and the ability to remove columns witha "X" button per column.`,
+
+  requires: ['foam.u2.view.ScrollDAOView'],
+  imports: [
+    'columns',
+    'removeColumn?',
+  ],
+
+  css: `
+    ^ {
+      display: flex;
+      min-height: 0;
+      flex-direction: column;
+    }
+
+    ^ col-heads {
+      z-index: 20;
+      flex-grow: 0;
+      flex-shrink: 0;
+      border-top: 1px solid #d0d0d0;
+      overflow-y: scroll;
+      background-color: rgb(248, 248, 248);
+      font-weight: bold;
+      height: 50px;
+      box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.14), 0 2px 1px -2px rgba(0, 0, 0, 0.2);
+    }
+
+    ^ col-head {
+      overflow: hidden;
+      white-space: nowrap;
+      text-overflow: ellipsis;
+      margin: auto 5px;
+      display: flex;
+      min-height: 0;
+      justify-content: space-between;
+    }
+
+    ^scroll-container {
+      position: relative;
+      flex-grow: 1;
+    }
+
+    ^ span, ^ .id {
+      overflow: hidden;
+      white-space: nowrap;
+      text-overflow: ellipsis;
+      margin: auto 5px;
+    }
+
+    ^ .id .overlay {
+      display: none;
+      position: absolute;
+      top: 0;
+      left: 0;
+    }
+
+    ^ li:hover .id {
+     color: transparent;
+    }
+
+    ^ li:hover .id .overlay {
+      height: 100%;
+      color: rgba(0, 0, 0, 0.8);
+      background-color: rgba(255, 255, 255, 0.6);
+      display: flex;
+      align-items: center;
+    }
+
+    ^icon {
+      z-index: 2;
+      display: inline-block;
+      padding: 1rem;
+    }
+
+    ^button {
+      flex-grow: 0;
+      flex-shrink: 0;
+      font-size: inherit;
+      background-color: inherit;
+      color: inherit;
+      padding: 0.5rem 0 0.5rem 0.5rem;
+      -webkit-user-select: none;
+      -moz-user-select: none;
+      -ms-user-select: none;
+      user-select: none;
+    }
+
+    ^button:hover {
+      background-color: rgba(255, 255, 255, 0.2);
+      cursor: pointer;
+    }
+  `,
+
+  classes: [
+    {
+      name: 'RowFormatter',
+      implements: ['foam.u2.RowFormatter'],
+
+      documentation: `Extend foam.u2.RowFormatter to support inversion of
+          control for rendering individual table cells.`,
+
+      imports: ['columns'],
+
+      methods: [
+        function format(data, opt_columns) {
+          const columns = opt_columns || this.columns;
+
+          let str = '';
+          for (let i = 0; i < columns.length; i++) {
+            const col = columns[i];
+            const value = data ? col.f(data) : undefined;
+            str += col.rawTableCellFormatter(data ? col.f(data) : undefined, data,
```

Comment:
What happens if there is no formatter?

Reply:
Good point. I added a safe default to `property.es6.js`.